docs/source/text_processing/g2p/g2p.rst

```diff
@@ -0,0 +1,203 @@
+
+The models take input data in `.json` manifest format, and there should be separate training and validation manifests.
+Each line of the manifest should be in the following format:
+
+.. code::
+
+  {"text_graphemes": "Swifts, flushed from chimneys.", "text": "ˈswɪfts, ˈfɫəʃt ˈfɹəm ˈtʃɪmniz."}
+
+Manifest fields:
+
+* ``text`` - name of the field in manifest_filepath for ground truth phonemes
+
+* ``text_graphemes`` - name of the field in manifest_filepath for input grapheme text
+
+The models can handle input with and without punctuation marks.
+
+To train ByT5 G2P model and evaluate it after at the end of the training, run:
+
+.. code::
+
+    python examples/text_processing/g2p/g2p_train_and_evaluate.py \
+        # (Optional: --config-path=<Path to dir of configs> --config-name=<name of config without .yaml>) \
+        model.train_ds.manifest_filepath="<Path to manifest file>" \
+        model.validation_ds.manifest_filepath="<Path to manifest file>" \
+        model.test_ds.manifest_filepath="<Path to manifest file>" \
+        trainer.devices=1 \
+        do_training=True \
+        do_testing=True
+
+Example of the config file: ``NeMo/examples/text_processing/g2p/conf/t5_g2p.yaml``.
+
+
+To train G2P-Conformer model and evaluate it after at the end of the training, run:
+
+.. code::
+
+    python examples/text_processing/g2p/g2p_train_and_evaluate.py \
+        # (Optional: --config-path=<Path to dir of configs> --config-name=<name of config without .yaml>) \
+        model.train_ds.manifest_filepath="<Path to manifest file>" \
+        model.validation_ds.manifest_filepath="<Path to manifest file>" \
+        model.test_ds.manifest_filepath="<Path to manifest file>" \
+        model.tokenizer.dir=<Path to pretrained tokenizer> \
+        model.tokenizer_grapheme.do_lower=False \
+        model.tokenizer_grapheme.add_punctuation=True \
+        trainer.devices=1 \
+
+        do_training=True \
+        do_testing=True
+
+Example of the config file: ``NeMo/examples/text_processing/g2p/conf/g2p_conformer_ctc.yaml``.
+
+
+To evaluate a pretrained G2P model, run:
+
+.. code::
+
+    python examples/text_processing/g2p/g2p_train_and_evaluate.py \
+        # (Optional: --config-path=<Path to dir of configs> --config-name=<name of config without .yaml>) \
+        pretrained_model="<Path to .nemo file or pretrained model name from list_available_models()>" \
+        model.test_ds.manifest_filepath="<Path to manifest file>" \
+        trainer.devices=1 \
+        do_training=False \
+        do_testing=True
+
+To run inference with a pretrained G2P model, run:
+
+.. code-block::
+
+    python g2p_inference.py \
+        pretrained_model=<Path to .nemo file or pretrained model name for G2PModel from list_available_models()>" \
+        manifest_filepath="<Path to .json manifest>" \
+        output_file="<Path to .json manifest to save prediction>" \
+        batch_size=32 \
+        num_workers=4
```

Comment:
Should we mention what the fields of the output .json file will look like?

Reply:
added